File: FE/src/app/models/Dish.ts
import {Category} from './Category';

export class Dish {
  id: number;
  name: string;
  description: string;
  amount: number;
  vote: number;
  voteCount: number;
  pickCount: number;
  category: Category;
  constructor(id, name) {
    this.id = id;
    this.name = name;
  }
  // constructor(id, name, description, amount, vote, voteCount, pickCount, category) {
  //   this.id = id;
  //   this.name = name;
  //   this.description = description;
  //   this.amount = amount;
  //   this.vote = vote;
  //   this.voteCount = voteCount;
  //   this.pickCount = pickCount;
  //   this.category = category;
  // }
}

File: BE/src/main/java/com/test/model/Dish.java
package com.test.model;

import javax.persistence.*;

@Entity
@Table(name = "dish")
public class Dish {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "dish_id")
    private Integer id;

    @Column(name = "dish_name")
    private String name;

    @Column(name = "dish_description")
    private String description;

    @Column(name = "dish_amount")
    private Integer amount;

    @Column(name = "dish_vote")
    private Integer vote;

    @Column(name = "dish_vote_count")
    private Integer voteCount;

    @Column(name = "dish_pick_count")
    private Integer pickCount;

    @ManyToOne
    @JoinColumn(name = "category_id", referencedColumnName = "category_id")
    private Category category;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Integer getAmount() {
        return amount;
    }

    public void setAmount(Integer amount) {
        this.amount = amount;
    }

    public Integer getVote() {
        return vote;
    }

    public void setVote(Integer vote) {
        this.vote = vote;
    }

    public Integer getVoteCount() {
        return voteCount;
    }

    public void setVoteCount(Integer voteCount) {
        this.voteCount = voteCount;
    }

    public Integer getPickCount() {
        return pickCount;
    }

    public void setPickCount(Integer pickCount) {
        this.pickCount = pickCount;
    }

    public Category getCategory() {
        return category;
    }

    public void setCategory(Category category) {
        this.category = category;
    }
}
